package.json

```diff
@@ -1,67 +1,69 @@
 {
   "name": "warcio",
-  "version": "1.6.0",
-  "main": "index.js",
-  "type": "module",
-  "license": "Apache-2.0",
-  "author": "Webrecorder Software, Ilya Kreymer",
-  "bin": {
-    "warcio.js": "./cli.js"
-  },
-  "exports": {
-    ".": "./index.js"
-  },
+  "version": "2.0.0",
+  "keywords": [
+    "WARC",
+    "web archiving"
+  ],
   "homepage": "https://github.com/webrecorder/warcio.js#readme",
+  "bugs": {
+    "url": "https://github.com/webrecorder/warcio.js/issues"
+  },
   "repository": {
     "type": "git",
     "url": "https://github.com/webrecorder/warcio.js.git"
   },
-  "bugs": {
-    "url": "https://github.com/webrecorder/warcio.js/issues"
-  },
-  "keywords": [
-    "WARC",
-    "web archiving"
+  "license": "Apache-2.0",
+  "author": "Webrecorder Software, Ilya Kreymer",
+  "sideEffects": [
+    "./src/polyfills.cjs"
   ],
-  "engines": {
-    "node": ">=18.0.0"
+  "type": "module",
+  "exports": {
+    "require": "./dist/index.cjs",
+    "default": "./dist/index.js"
   },
+  "main": "dist/index.cjs",
+  "module": "dist/index.js",
+  "source": "src/index.ts",
+  "types": "dist/index.d.ts",
+  "bin": "./dist/cli.js",
   "files": [
     "/src",
-    "/dist",
-    "index.js",
-    "cli.js"
+    "/dist"
   ],
-  "dependencies": {
-    "hi-base32": "^0.5.0",
-    "pako": "^1.0.11",
-    "uuid-random": "^1.3.0",
-    "yargs": "^15.3.1"
-  },
   "scripts": {
-    "test": "c8 ava",
-    "coverage": "codecov",
-    "build": "PRODUCTION=1 rollup -c",
-    "lint": "eslint ./src/ ./test/ *.js"
+    "build": "tsup src/index.ts src/cli.ts",
+    "lint": "eslint 'src/**/*.ts'",
+    "test": "npx jest --collectCoverage"
   },
-  "ava": {
-    "concurrency": 1,
-    "verbose": true,
-    "serial": true,
-    "files": [
-      "test/*.js"
-    ]
+  "dependencies": {
+    "hi-base32": "^0.5.1",
+    "pako": "^1.0.11",
+    "uuid-random": "^1.3.2",
+    "yargs": "^17.6.2"
   },
   "devDependencies": {
-    "@rollup/plugin-commonjs": "^11.1.0",
-    "@rollup/plugin-node-resolve": "^7.1.3",
-    "ava": "^4.3.3",
-    "c8": "^7.12.0",
-    "codecov": "^3.8.3",
+    "@types/eslint": "^7.29.0",
+    "@types/jest": "^29.2.3",
+    "@types/node": "^18.11.9",
+    "@types/pako": "^1.0.0",
+    "@types/stream-buffers": "^3.0.4",
+    "@types/yargs": "^17.0.13",
+    "@typescript-eslint/eslint-plugin": "^5.43.0",
+    "@typescript-eslint/parser": "^5.43.0",
+    "cross-fetch": "^3.1.5",
     "eslint": "^7.24.0",
-    "rollup": "^2.6.1",
-    "rollup-plugin-license": "^2.1.0",
-    "rollup-plugin-terser": "^5.3.0",
-    "stream-buffers": "^3.0.2"
+    "eslint-plugin-deprecation": "^1.3.3",
+    "jest": "^29.3.1",
+    "jest-expect-message": "^1.1.3",
+    "pako": "^1.0.0",
+    "stream-buffers": "^3.0.2",
+    "ts-jest": "^29.0.3",
+    "tsup": "^6.5.0",
+    "typescript": "^4.9.3"
+  },
+  "engines": {
```

Comment:
In 1.6.1, removed the >=18.0.0 requirement, as still need to be able to bundle when running in older versions of node, and that works well.